lib/src/rules/prefer_foreach.dart

```diff
@@ -121,6 +116,29 @@ class _PreferForEachVisitor extends SimpleAstVisitor {
   }
 }
 
+class _ReferenceFinder extends GeneralizingAstVisitor {
+  bool found = false;
+  final LocalVariableElement? element;
+  _ReferenceFinder(this.element);
+
+  bool references(Expression target) {
+    if (target.canonicalElement == element) return true;
```

Comment:
This test will be repeated in `visitNode`. If there isn't a performance benefit from not visiting the node (which there might well be if we expect this to return immediately often enough), then consider removing it.

Reply:
It's a good point. I'm inclined to short-circuit visiting wherever possible but I haven't measured this...  I'll look a bit closer.  Thanks!